lib/govuk_publishing_components/presenters/public_layout_helper.rb

```diff
@@ -46,17 +46,48 @@ class PublicLayoutHelper
       attr_reader :footer_navigation, :footer_meta, :cookie_banner_data
 
       def initialize(local_assigns)
+        @footer_links_count = generate_ga4_footer_links_count(local_assigns)
         @footer_navigation = local_assigns[:footer_navigation] || navigation_link_generation_from_locale(I18n.t("components.layout_footer.navigation_links"))
-        @footer_meta = local_assigns[:footer_meta] || { items: add_data_attributes_to_links(FOOTER_META[:items], "supportLink") }
+        @footer_meta = local_assigns[:footer_meta] || { items: add_data_attributes_to_links(FOOTER_META[:items], "supportLink", { section: "Support links", index_section: 3 }) }
         @cookie_banner_data = local_assigns[:cookie_banner_data] || {}
       end
 
+      def generate_ga4_footer_links_count(local_assigns)
+        footer_links_count = 0
+
+        footer_nav = local_assigns[:footer_navigation] || I18n.t("components.layout_footer.navigation_links")
+
+        footer_nav.each do |column|
+          if column[:items]
+            footer_links_count += column[:items].length
+          elsif column[:menu_contents]
+            footer_links_count += column[:menu_contents].length
+          end
+        end
+
+        footer_meta = local_assigns[:footer_meta] || FOOTER_META
+        footer_links_count += footer_meta[:items].length
+
+        extra_footer_links = 2 # Amount of links to add to @footer_links_count. This accounts for the OGL link and Crown Copyright link.
+
+        footer_links_count += extra_footer_links
+        footer_links_count
+      end
+
+      def get_footer_links_count
+        @footer_links_count
+      end
+
       def navigation_link_generation_from_locale(links)
         links.each_with_index.map do |menu, i|
+          ga4_data = {
+            section: menu[:title],
+            index_section: i + 1,
+          }
           {
             title: menu[:title],
             columns: footer_navigation_columns[i],
-            items: add_data_attributes_to_links(menu[:menu_contents], footer_track_actions[i]),
+            items: add_data_attributes_to_links(menu[:menu_contents], footer_track_actions[i], ga4_data),
```

Comment:
We are passing the `section` and `index_section` which we calculated in this function downwards towards the `add_data_attributes_to_links` function.